pkg/autodiscovery/providers/kubecontainer.go

```diff
@@ -0,0 +1,315 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !serverless
+// +build !serverless
+
+package providers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/common/utils"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/providers/names"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/telemetry"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+)
+
+// KubeContainerConfigProvider implements the ConfigProvider interface for both kubelet and containers
+type KubeContainerConfigProvider struct {
```

Comment:
Not in love with the name, gives the impression that this is only about kubernetes containers (which it's not, it also runs in ECS). It can stay for now, but I think it would be useful to document in code that this will replace the other providers, and my suggestion is that this gets renamed to `container` when all is said and done.

Reply:
I like changing to just `container` once this is fully deployed.

I'll continue with this name for now and comment in the code that it will supersede both the `KubeletConfigProvider` and the `ContainerConfigProvider` once cca-in-ad is fully deployed (and be renamed to `container`)